src/haxelib/api/GlobalScope.hx

```diff
@@ -186,15 +187,18 @@ class GlobalScope extends Scope {
 			addLine('-D ${info.name}=${info.version}');
 
 			// add dependencies to stack
-			final dependencies = info.dependencies.toArray();
+			final dependencies = info.dependencies.extractDataArray();
 
 			while (dependencies.length > 0) {
 				final dependency = dependencies.pop();
 				stack.add({
 					library: ProjectName.ofString(dependency.name),
 					version: // TODO: maybe check the git/hg commit hash here if it's given?
-						if (dependency.version == DependencyVersion.DEFAULT) null
-						else Version.ofString(dependency.version)
+						switch dependency.versionData {
+							case None: null;
```

Comment:
Hmm, most of the recent changes went from `null` stuff to `Option` stuff, but here it's the other way around. If `version` is optional, we might want to turn that into an Option as well. But that's also not very urgent.